tests/test_choices.py

```diff
@@ -1,402 +1,402 @@
+    return dataclass(**_DATACLASS_KWARGS)(mock_apply(cls))
+
+
+def mock_apply(cls):
+    """Only mock apply, without applying dataclass. For testing deprecated API."""
+    orig_init = cls.__init__
+
+    def __init__(self, *args, **kwargs):
+        self.apply = Mock()
+        orig_init(self, *args, **kwargs)
+
+    cls.__init__ = __init__
+    return cls
+
+
+@prep_style_class
+class Style(BaseStyle):
+    none: str = validated_property(choices=Choices(NONE, REBECCAPURPLE), initial=NONE)
+    allow_string: str = validated_property(
+        choices=Choices(string=True), initial="start"
+    )
+    allow_integer: int = validated_property(choices=Choices(integer=True), initial=0)
+    allow_number: float = validated_property(choices=Choices(number=True), initial=0)
+    allow_color: str = validated_property(
+        choices=Choices(color=True), initial="goldenrod"
+    )
+    values: str = validated_property(choices=Choices("a", "b", NONE), initial="a")
+    multiple_choices: str | float = validated_property(
+        choices=Choices("a", "b", NONE, number=True, color=True),
+        initial=None,
+    )
+    string_symbol: str = validated_property(choices=Choices(TOP, NONE))
+
+
+with catch_warnings():
+    filterwarnings("ignore", category=DeprecationWarning)
+
+    @mock_apply
+    class DeprecatedStyle(BaseStyle):
+        pass
+
+    DeprecatedStyle.validated_property(
+        "none", choices=Choices(NONE, REBECCAPURPLE), initial=NONE
+    )
+    DeprecatedStyle.validated_property(
+        "allow_string", choices=Choices(string=True), initial="start"
+    )
+    DeprecatedStyle.validated_property(
+        "allow_integer", choices=Choices(integer=True), initial=0
+    )
+    DeprecatedStyle.validated_property(
+        "allow_number", choices=Choices(number=True), initial=0
+    )
+    DeprecatedStyle.validated_property(
+        "allow_color", choices=Choices(color=True), initial="goldenrod"
+    )
+    DeprecatedStyle.validated_property(
+        "values", choices=Choices("a", "b", NONE), initial="a"
+    )
+    DeprecatedStyle.validated_property(
+        "multiple_choices",
+        choices=Choices("a", "b", NONE, number=True, color=True),
+        initial=None,
+    )
+    DeprecatedStyle.validated_property("string_symbol", choices=Choices(TOP, NONE))
+
+
+@pytest.mark.parametrize("StyleClass", [Style, DeprecatedStyle])
+def test_none(StyleClass):
+    style = StyleClass()
+    assert style.none == NONE
+
+    with pytest.raises(ValueError):
+        style.none = 10
+
+    with pytest.raises(ValueError):
+        style.none = 3.14159
+
+    with pytest.raises(ValueError):
+        style.none = "#112233"
+
+    with pytest.raises(ValueError):
+        style.none = "a"
+
+    with pytest.raises(ValueError):
+        style.none = "b"
+
+    # Set the property to a different explicit value
+    style.none = REBECCAPURPLE
+    assert style.none == REBECCAPURPLE
```

Comment:
Oh, I made a critical misread. Somehow I thought I read the default `check_mock` as `False` and thought it wasn't getting called. Will re-add.

Reply:
This also lead me to notice I applied the dataclass and mock_apply decorators in the wrong order. It didn't cause any tests to fail, but that's because of the backwards compatibility. I've swapped them now so dataclass is applied first, it generates its own init (if >= 3.10), and *then* apply is mocked.